### PR TITLE
DRY up case_config_ui.js

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
This DRYs up the code duplicated between caseTransaction userCaseTransaction in case_config_ui.js

Definitely review by commit, although some of the commit diffs are confusing because of the amount of code moved. All changes in this PR are pretty mechanical: moving code and occasionally turning something into a parameter or updating a reference.

##### RISK ASSESSMENT / QA PLAN
Sanity tested locally. This is some of our most complicated javascript, and it has no automated tests. My (hopefully) saving grace is that these changes are mechanical.

### CODE REVIEW

**Comment by orangejenny** [comment]
Ignoring stickler.

**Review by millerdev — APPROVED** [review]

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js [hunk 1]
```diff
@@ -504,36 +505,6 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                 return 'commcare-user';
             };
 
-            if (self.case_preload) {
```
Comment (calellowitz) [inline corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js]: what happens to this?

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js [hunk 2]
```diff
@@ -505,35 +505,6 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                 return 'commcare-user';
             };
 
-            self.setRequired = function (required) {
```
Comment (calellowitz) [inline corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js]: same question I have about everything else. was this not used?

**Review by calellowitz — APPROVED** [review]
Left a few questions but generally looks good.

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js [hunk 3]
```diff
@@ -496,48 +498,12 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
...
+            var self = baseTransaction(userCaseTransactionMapping, self.caseConfig.saveUsercaseButton, 'User Case Management', data, caseConfig, true);
 
-            ko.mapping.fromJS(data, userCaseTransactionMapping(self), self);
             self.case_type = function () {
                 return 'commcare-user';
             };
 
-            // link self.case_name to corresponding path observable
-            // in case_properties for convenience
-            try {
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js]: This wasn't removed, it's just a weird diff. Check out the full diff.

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js [hunk 4]
```diff
@@ -319,36 +319,29 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
             return baseMapping(model, ['case_properties', 'case_preload']);
         };
 
-        var baseTransaction = function (data, caseConfig, hasPrivilege) {
+        var baseTransaction = function (mapping, saveButton, analyticsAction, data, caseConfig, hasPrivilege) {
             var self = {};
+            ko.mapping.fromJS(data, mapping(self), self);
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js:324]: `self` is the target that gets updated: `fromJS` takes a plain object `data`, transforms its properties into observables using the given mapping, and then assigns the results to the target. https://knockoutjs.com/documentation/plugins-mapping.html (ctrl+F "third parameter")

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js [hunk 5]
```diff
@@ -504,18 +504,6 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                 return 'commcare-user';
             };
 
-            self.propertyCounts = ko.computed(function () {
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js]: It's another weird diff. Something about de-duping big blocks of code near each other is confusing for GitHub.

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js]: weird diff

File: corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js]: Pretty sure this is a weird diff, but if you look at the full diff and still have a question, let me know.

**Pull request merged**